upstream-sha256: f7af4802c2fd9f1644642755a00555b84ba51ffba028263ed8ae4f13bdca6362
actual-sha256: ba3eae333b87ef3057c6d19bc4a17b59b09fe8157eda0628581eafc1e01aed5e
--- upstream.expected
+++ tolk.actual
@@ -15,7 +15,7 @@
       float val4 = (*(data4_16+(Lidx3+4)));
       float val5 = (*(data4_16+(Lidx3+8)));
       float val6 = (*(data4_16+Lidx3));
-      *(data0_16+(alu0+Lidx3)) = ((*(buf0+0))+(val0[0]*val6)+(val0[1]*val4)+(val0[2]*val5)+(val0[3]*val3));
+      *(data0_16+(alu0+Lidx3)) = ((*(buf0+0))+(0.0f+(val0[0]*val6)+(val0[1]*val4)+(val0[2]*val5)+(val0[3]*val3)));
     }
   }
 }
@@ -37,7 +37,7 @@
       float val4 = (*(data4_16+(Lidx3+4)));
       float val5 = (*(data4_16+(Lidx3+8)));
       float val6 = (*(data4_16+Lidx3));
-      *(data0_16+(alu0+Lidx3)) = (2.0f*((*(buf0+0))+(val0[0]*val6)+(val0[1]*val4)+(val0[2]*val5)+(val0[3]*val3)));
+      *(data0_16+(alu0+Lidx3)) = (2.0f*((*(buf0+0))+(0.0f+(val0[0]*val6)+(val0[1]*val4)+(val0[2]*val5)+(val0[3]*val3))));
     }
   }
 }
@@ -51,7 +51,7 @@
       int alu1 = (alu0+Lidx2);
       float val1 = (*(data1_16+alu1));
       float4 val2 = (*((float4*)((data3_16+(Lidx2<<2)))));
-      *(data0_16+alu1) = ((2.0f*val1)+(val0[0]*val2[0])+(val0[1]*val2[1])+(val0[2]*val2[2])+(val0[3]*val2[3]));
+      *(data0_16+alu1) = ((2.0f*val1)+(0.0f+(val0[0]*val2[0])+(val0[1]*val2[1])+(val0[2]*val2[2])+(val0[3]*val2[3])));
     }
   }
 }
@@ -75,7 +75,7 @@
       float val7 = (*(data4_16+(Lidx3+4)));
       float val8 = (*(data4_16+(Lidx3+8)));
       float val9 = (*(data4_16+Lidx3));
-      *(data0_16+((Lidx2<<2)+Lidx3)) = ((*(buf0+0))+(val3*val9)+(val1*val7)+(val2*val8)+(val0*val6));
+      *(data0_16+((Lidx2<<2)+Lidx3)) = ((*(buf0+0))+(0.0f+(val3*val9)+(val1*val7)+(val2*val8)+(val0*val6)));
     }
   }
 }
@@ -99,7 +99,7 @@
       float val7 = (*(data4_16+(Lidx3+4)));
       float val8 = (*(data4_16+(Lidx3+8)));
       float val9 = (*(data4_16+Lidx3));
-      *(data0_16+((Lidx2<<2)+Lidx3)) = ((*(buf0+0))+(val3*val9)+(val1*val7)+(val2*val8)+(val0*val6));
+      *(data0_16+((Lidx2<<2)+Lidx3)) = ((*(buf0+0))+(0.0f+(val3*val9)+(val1*val7)+(val2*val8)+(val0*val6)));
     }
   }
 }
@@ -111,7 +111,7 @@
     float4 val0 = (*((float4*)((data1_16+alu0))));
     for (int Lidx2 = 0; Lidx2 < 4; Lidx2++) {
       float4 val1 = (*((float4*)((data2_16+(Lidx2<<2)))));
-      *(data0_16+(alu0+Lidx2)) = ((val0[0]*val1[0])+(val0[1]*val1[1])+(val0[2]*val1[2])+(val0[3]*val1[3]));
+      *(data0_16+(alu0+Lidx2)) = (0.0f+(val0[0]*val1[0])+(val0[1]*val1[1])+(val0[2]*val1[2])+(val0[3]*val1[3]));
     }
   }
 }
